### ISSUE TITLE
Fails when there's some additional string in output

### ISSUE THREAD

**Issue by JJ** [perl6/tap-harness6#33]
When there's something like this:

```
    ok 4 - routine name input
stuff
    ok 5 - identifier (sub) input
```

it fails as indicated in #32 
prove didn't fail.

**Comment by Leont** [comment]
Yeah, that is totally an invalid TAP subtest for obvious reasons. `prove` happens not to notice because it completely ignores subtests.

I'm not sure why one would expect random output in the middle of any data format to work. TAP is traditionally quite tolerant of such things, but it's not very future-compatible.

Apparently a lot of tests are sloppily written because the traditional consumer was so tolerant. But quite frankly the options are:
1. Fix test output to be proper TAP
2. Give up on subtests (at least on the harness side). This is what `prove` does. `prove6` has a `--loose` argument that does this as well.

If people really want it I can switch it around, but it does make me sad

**Comment by JJ** [comment]
So I guess we should make #25 work... Or close it if it already does.

**Comment by tbrowder** [comment]
I originally wanted the "--loose" option because of the probably unexpected behavior for the messages I wanted to see when I was writing LOTS of tests. Maybe I never found a good tutorial on how the whole test harness in Rakudo all fits together. I saw the option in the code but not described in the README.

**Comment by Leont** [comment]
Now that I think of it, I think it's possible to automatically downgrade from strict to loose. I should investigate this.

**Comment by robertlemmen** [comment]
I keep stumbling into the same problems as well, here are a few thoughts now:
- the software under test may emit output, kinda like "diag" but it should not need to rely on the test module to do this. so this kind of debug out put could be prefixed with "#" to make it a TAP-comment
- however, within a subtest, TAP requires the comment to be indented like the rest of the subtest, otherwise it upsets the plan. I can't see why, and I can't see that behaviour in the grammar, but it's easy to reproduce. perhaps TAP should accept comments as comments  no matter what the indentation?
- I was wrestling with this for a while, but now I am a happier man by simply sending my diagnostics to $*ERR where TAP doesn't care :)
